tuxemon/states/journal/journal.py

```diff
@@ -0,0 +1,145 @@
+# SPDX-License-Identifier: GPL-3.0
+# Copyright (c) 2014-2023 William Edwards <shadowapex@gmail.com>, Benjamin Bean <superman2k5@gmail.com>
+from __future__ import annotations
+
+from functools import partial
+from typing import Any
+
+import pygame_menu
+from pygame_menu import locals
+from pygame_menu.locals import POSITION_CENTER
+
+from tuxemon import prepare, tools
+from tuxemon.db import MonsterModel, SeenStatus
+from tuxemon.locale import T
+from tuxemon.menu.menu import BACKGROUND_COLOR, PygameMenuState
+from tuxemon.menu.theme import get_theme
+from tuxemon.session import local_session
+from tuxemon.states.journal.journal_info import JournalInfoState
+
+MAX_PAGE = 20
+
+
+def fix_width(screen_x: int, pos_x: float) -> int:
+    """it returns the correct width based on percentage"""
+    value = round(screen_x * pos_x)
+    return value
+
+
+def fix_height(screen_y: int, pos_y: float) -> int:
+    """it returns the correct height based on percentage"""
+    value = round(screen_y * pos_y)
+    return value
+
+
+class JournalState(PygameMenuState):
+    """Shows journal (screen 2/3)."""
+
+    def add_menu_items(
+        self,
+        menu: pygame_menu.Menu,
+        monsters: list[MonsterModel],
+    ) -> None:
+        width = menu._width
+        height = menu._height
+        menu._column_max_width = [
+            fix_width(width, 0.35),
+            fix_width(width, 0.35),
+        ]
+
+        monsters = sorted(monsters, key=lambda x: x.txmn_id)
+
+        player = local_session.player
+        for mon in monsters:
+            if mon.slug in player.tuxepedia:
+                label = str(mon.txmn_id) + ". " + T.translate(mon.slug).upper()
+                if player.tuxepedia[mon.slug] == SeenStatus.seen:
+                    menu.add.button(
+                        label,
+                        partial(self.client.push_state, JournalInfoState(mon)),
```

Comment:
Using `partial` and `push_state` with an instance `State` should be avoided.  The reason being is that since the `JournalInfoState` is being created here, *and* is is optional that the play presses the button, we could be wasting CPU time and memory creating the `JournalInfoState`, loading graphics, etc.  In this instance, you should pass the name of the state as a string, and the arguments and kwargs.
Doing it this way ensure that the next state is only created when needed.  In some situations, its fine to `push_state` with an instanced State object if we 100% know that the state needs to change.  But its part of a optional button, so we should avoid creating it if the player might not open it.

Reply:
ok, I'll keep it in mind, fixed